src/divider.jsx

```diff
@@ -30,35 +33,34 @@ const defaultProps = {
 
 let Divider = (props) => {
   const {
-    inset,
-    muiTheme,
-    style,
+    _styleRoot,
     ...other,
   } = props;
 
-  const {
-    prepareStyles,
-  } = muiTheme;
-
-  const styles = {
-    root: {
-      margin: 0,
-      marginTop: -1,
-      marginLeft: inset ? 72 : 0,
-      height: 1,
-      border: 'none',
-      backgroundColor: muiTheme.rawTheme.palette.borderColor,
-    },
-  };
-
   return (
-    <hr {...other} style={prepareStyles(Object.assign({}, styles.root, style))} />
+    <hr {...other} style={_styleRoot} />
   );
 };
 
 Divider.displayName = 'Divider';
 Divider.propTypes = propTypes;
 Divider.defaultProps = defaultProps;
+
+const rootStyleSelector = createSelector('_styleRoot', [
+  (props) => props.inset,
+  (props) => props.muiTheme.rawTheme.palette.borderColor,
+  (props) => props.muiTheme.style,
+  (props) => props.muiTheme.prepareStyles,
+], (inset, borderColor, style, prepareStyles) => prepareStyles(Object.assign({
+  margin: 0,
+  marginTop: -1,
+  marginLeft: inset ? 72 : 0,
+  height: 1,
+  border: 'none',
+  backgroundColor: borderColor,
+}), style));
```

Comment:
Nope, that's why we're in different timezones! Believe me in about 9 hours you're going to be commenting on my silly code :laughing: